### PR TITLE
Simplify the logic of configuring integration with CloudKMS.

### PR DESCRIPTION
**Pull request by immutableT**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In the past, we supported an option where users could supply EncryptionConfig that was NOT for CloudKMS (ex. AES-GCM or SecretBox providers). This is no longer the case, we only accept configs for CloudKMS since the other providers have some inherent security issues related to how they store their keys. This PR reflects this fact by removing the need to check for ENABLE_INTEGRATION_WITH_CLOUDK_KMS var, since CloudKMS is the only supported integration mechanism.

1. Is there an ENCRYPTION_PROVIDER_CONFIG variable and if yes
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cluster/gce/gci/apiserver_manifest_test.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ import (
 	"strings"
 	"testing"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (cjcullen) [inline cluster/gce/gci/apiserver_manifest_test.go]: This shouldn't be necessary, right?

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -2124,8 +2123,8 @@ function setup-etcd-encryption {
   echo "${ENCRYPTION_PROVIDER_CONFIG}" | base64 --decode > "${encryption_provider_config_path}"
   kube_api_server_params+=" --encryption-provider-config=${encryption_provider_config_path}"
 
-  default_encryption_provider_config_vol=$(echo "{ \"name\": \"encryptionconfig\", \"hostPath\": {\"path\": \"${encryption_provider_config_path}\", \"type\": \"File\"}}" | base64 | tr -d '\r\n')
```
Comment (cjcullen) [inline cluster/gce/gci/configure-helper.sh]: Does this leave newlines in the output if you yank out the `|tr -d '\r\n'`?

Comment (immutableT) [reply cluster/gce/gci/configure-helper.sh]: Yes. 
~$ echo "foo" | base64 
Zm9vCg==
~$ 

The "tr -d" was there to make this code run properly on Macs - I don't think this is a requirement anymore.

File: cluster/gce/gci/apiserver_manifest_test.go (hunk 1 above)
Comment (immutableT) [reply cluster/gce/gci/apiserver_manifest_test.go]: I don't think it is (IJ added this automatically), but I do see that we do this in a lot of places in k8s.

File: cluster/gce/gci/configure-helper.sh (hunk 2 above)
Comment (cjcullen) [reply cluster/gce/gci/configure-helper.sh]: If your input is long enough, you will generate a base64 string that won't decode properly:

```
$ printf 'A%.0s' {1..100} | base64
QUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFB
QUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQUFBQQ==

$ foo=$(printf 'A%.0s' {1..100} | base64)

$ echo $foo | base64 -d
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAbase64: invalid input
```
`base64 -w` fixes this on Linux, but the Darwin version doesn't have the same flag.

File: cluster/gce/gci/apiserver_manifest_test.go (hunk 1 above)
Comment (cjcullen) [reply cluster/gce/gci/apiserver_manifest_test.go]: Interesting. I do see it in a ton of places.

File: cluster/gce/gci/configure-helper.sh (hunk 2 above)
Comment (immutableT) [reply cluster/gce/gci/configure-helper.sh]: Thank you for catching this.
Added "-w0" flag.

File: cluster/gce/gci/apiserver_manifest_test.go (hunk 1 above)
Comment (cjcullen) [reply cluster/gce/gci/apiserver_manifest_test.go]: For posterity, this is due to how goimports makes assumptions about `foo/v1` being a go modules version 1 of the `foo` package (https://go-review.googlesource.com/c/tools/+/122616/).

File: cluster/gce/gci/configure-helper.sh (hunk 2 above)
Comment (cjcullen) [reply cluster/gce/gci/configure-helper.sh]: Is there a reason not to just leave the `|tr -d '\r\n'`? Did it break something?

Comment (immutableT) [reply cluster/gce/gci/configure-helper.sh]: No, this does not break anything.
Removing only to simplify .
As I mentioned, this was added to support Macs. I doubt we still need this.

**Pull request closed without merging**